pyiceberg/table/__init__.py

```diff
@@ -1975,6 +1948,52 @@ def _commit(self) -> UpdatesAndRequirements:
         """Apply the pending changes and commit."""
         return self._updates, self._requirements
 
+    def _commit_if_ref_updates_exist(self) -> None:
+        self._transaction._apply(*self._commit(), commit_transaction_now=False)
+        self._updates, self._requirements = (), ()
+
+    def _stage_main_branch_snapshot_ref(self, snapshot_id: int) -> None:
+        update, requirement = self._set_ref_snapshot(
+            snapshot_id=snapshot_id, ref_name=MAIN_BRANCH, type=str(SnapshotRefType.BRANCH)
+        )
+        self._updates += update
+        self._requirements += requirement
+
+    def _set_ref_snapshot(
+        self,
+        snapshot_id: int,
+        ref_name: str,
+        type: str,
+        max_ref_age_ms: Optional[int] = None,
+        max_snapshot_age_ms: Optional[int] = None,
+        min_snapshots_to_keep: Optional[int] = None,
+    ) -> UpdatesAndRequirements:
+        """Update a ref to a snapshot.
+
+        Returns:
+            The updates and requirements for the set-snapshot-ref staged
+        """
+        updates = (
+            SetSnapshotRefUpdate(
+                snapshot_id=snapshot_id,
+                ref_name=ref_name,
+                type=type,
+                max_ref_age_ms=max_ref_age_ms,
+                max_snapshot_age_ms=max_snapshot_age_ms,
+                min_snapshots_to_keep=min_snapshots_to_keep,
+            ),
+        )
+        requirements = (
+            AssertRefSnapshotId(
+                snapshot_id=self._transaction.table_metadata.refs[ref_name].snapshot_id
+                if ref_name in self._transaction.table_metadata.refs
+                else None,
+                ref=ref_name,
+            ),
+        )
+
+        return updates, requirements
```

Comment:
Instead of return `updates` and `requirements`, how about directly updating the field here
```python
self._updates += update
self._requirements += requirement
```
This way we do not need to repeat these two lines and all APIs will end with a `self._set_snapshot_ref(...)`

Reply:
making this change lead to more changes like 
- converting `_set_ref_snapshot` unit test into an integration test, since we can't commit() in a unit test.
- for the test I had to let `_set_ref_snapshot` return ManageSnapshots(), so all APIs end with 
`return self._set_snapshot_ref(...)`